app/tool/daq.cxx

```diff
@@ -666,6 +666,65 @@ static void daq_debug_trigger_timein(Target* tgt) {
       "Want to try another set of timing parameters?", false));
 }
 
+template <class EventPacket>
+std::unique_ptr<DecodeAndWriteToCSV<EventPacket>> make_all_channel_writer(
+    const std::string& file_name) {
+  return std::make_unique<DecodeAndWriteToCSV<EventPacket>>(
+      file_name,
+      [](std::ofstream& f) {
+        f << std::boolalpha;
+        f << EventPacket::to_csv_header << '\n';
+      },
+      [](std::ofstream& f, const EventPacket& ep) { ep.to_csv(f); }, 2);
+}
+
+std::unique_ptr<DAQRunConsumer> make_all_channel_writer(
+    Target::DaqFormat f, const std::string& file_name) {
+  switch (f) {
+    case Target::DaqFormat::ECOND_SW_HEADERS:
+      return make_all_channel_writer<
+          pflib::packing::MultiSampleECONDEventPacket>(file_name);
+      break;
+    case Target::DaqFormat::SIMPLEROC:
+      return make_all_channel_writer<pflib::packing::SingleROCEventPacket>(
+          file_name);
+      break;
+    default:
+      PFEXCEPTION_RAISE("NoDecode",
+                        "Unable to decode format currently configured.");
+      break;
+  }
+}
```

Comment:
This is the only thing I really want to keep. The other stuff is probably (hopefully) not necessary.